tests/test_client_helpers.py

```diff
@@ -235,3 +236,18 @@ def test_rev_parse_error_on_parent_does_not_exist(
         client_helpers.rev_parse(
             client=fs.client, repository=repository, ref=temp_branch, parent=non_existent_parent
         )
+
+
+def test_delete_branch(fs: LakeFSFileSystem, repository: str) -> None:
```

Comment:
Maybe add a case covering the error case (could just be a `with pytest.raises(ApiException): ...`), so that we know we raise the correct exceptions